### ISSUE TITLE
executeDeepLink open Apps instead of the actual Tab

### ISSUE THREAD

**Issue by Nootvar** [OfficeDev/microsoft-teams-library-js#670]
I would like, from an SPFX tab, open another SPFX tab in another team.

I can recover the `webUrl` of the tab via graph without issues.
Opening the `webUrl` in my browser open the tab as expected.
Using `executeDeepLink` to open the `webUrl` open the Apps Catalog of Teams with the installation popup for the SPFX app.

The webUrl` returned by graph is already properly encoded so it does not seems the source of the issue.
It seems that `executeDeepLink` is not reacting well to links aiming at application tabs while opening the same link in you browser works as expected.

**Comment by v-chetsh** [comment]
@Nootvar - Could you please share your deeplink Url and documentation which you are referring, so that we can check it from our end.

**Comment by Nootvar** [comment]
Of course, to add to the context here is the code I'm using in the SPFX App : 
````ts
public async navigateToAppTab(groupId: string, appId: string) {
    let tabInfo = await this.searchTab(groupId, appId);
    console.log(tabInfo.tabUrl);
    this.context.sdks.microsoftTeams.teamJs.executeDeepLink(tabInfo.tabUrl);
}

private async searchTab(groupId: string, appId: string) {
    let client = await this.context.msGraphClientFactory.getClient();
    let response = await client.api(`/teams/${groupId}/channels`).select('id').get();
    for (let channel of response.value) {
        response = await client.api(`/teams/${groupId}/channels/${channel.id}/tabs`).select('webUrl').get();
        for (let tab of response.value)
            if (tab.webUrl.includes(appId))
                return {tabUrl: tab.webUrl, channelId: channel.id};
    }
    return null;
}
````

So I can call `navigateToAppTab` where `groupId` is the id of the Team and `appId` is the id of the application (again here it's an SPFX application).
`searchTab` find the tab without any issue and return the id of the channel and the url of the tab.
The `tabUrl` (`webUrl` of the Graph object) recovered is the following : https://teams.microsoft.com/l/entity/9dc68128-88bf-4024-a275-7d049fbc7ac3/_djb2_msteams_prefix_96bbe562-5a47-4e15-a84f-0d2615883c7b?label=Information+de+dossier&context=%7b%0d%0a++%22canvasUrl%22%3a+%22https%3a%2f%2fthenrbgroup.sharepoint.com%2fteams%2fgfc_DossierPrDEMOVendredi17-MOB%2f_layouts%2f15%2fTeamsLogon.aspx%3fSPFX%3dtrue%26dest%3d%2fteams%2fgfc_DossierPrDEMOVendredi17-MOB%2f_layouts%2f15%2fteamshostedapp.aspx%253Flist%3d49e5554c-3628-494b-bf86-3eca084007da%2526id%3d1%2526webPartInstanceId%3d43cc7280-1153-4516-80cc-5a19d177f09c%22%2c%0d%0a++%22channelId%22%3a+%2219%3a7DSBv4H7fKAXrn5D72srpkMrvSwVqQb0qz7EKo6E9001%40thread.tacv2%22%2c%0d%0a++%22subEntityId%22%3a+null%0d%0a%7d&groupId=7c541703-b4d3-4021-9281-51f1fe83713b&tenantId=b6c46f52-625c-414b-aca2-c4a4f0e7bd20

Opening this link directly in a browser works fine and open the tab as expected.

As per the documentation [Create deep links](https://docs.microsoft.com/en-us/microsoftteams/platform/concepts/build-and-test/deep-links), it seems a complicated but fine link. But doing `this.context.sdks.microsoftTeams.teamJs.executeDeepLink('{the link given above}');` open the Teams App Catalog page, ignoring the rest of the parameters of the link and only considering the application id.

**Comment by Prasad-MSFT** [comment]
@Nootvar , Could you please try by getting the tab url as below instead of getting it by graph API.
![SS111](https://user-images.githubusercontent.com/62237649/135268442-b6340ca3-9e07-484a-b37a-eef62691891a.PNG)
It worked at our end.The url will be in below format:
`https://teams.microsoft.com/l/entity/da6eaf79-08f0-47d8-9e71-179a8589e546/_djb2_msteams_prefix_238113063?context=%7B%22subEntityId%22%3Anull%2C%22channelId%22%3A%2219%3Aac7b9c53a099498f9e08679e58f1f7fc%40thread.tacv2%22%7D&groupId=1b85f4f3-fa01-46f4-bb62-201225dce9e4&tenantId=36a708ef-700d-4d60-9de0-0a5f7b7693df`

**Comment by Nootvar** [comment]
Indeed the link I can get from Teams does work with `executeDeepLink`.

However I need to generate the link programmatically and the link format you have given me is not the same as the one I get : `https://teams.microsoft.com/l/entity/9dc68128-88bf-4024-a275-7d049fbc7ac3/_djb2_msteams_prefix_1132819653?context=%7B%22subEntityId%22%3Anull%2C%22channelId%22%3A%2219%3A7DSBv4H7fKAXrn5D72srpkMrvSwVqQb0qz7EKo6E9001%40thread.tacv2%22%7D&groupId=7c541703-b4d3-4021-9281-51f1fe83713b&tenantId=b6c46f52-625c-414b-aca2-c4a4f0e7bd20`

I can recover almost all the ids in this link to construct it myself, except `_djb2_msteams_prefix_1132819653`. It's not the `entityId` since looking at the context of the actual tab, the `entityId` is the following : `sharepointtab_543cf38c-f64e-4408-bb47-21240f6dcfbe`.

I've tried to create the same link as the one you recommend but I cannot figure out where to get the `groupChatId`. In the context it is always `null`. The link I recover from `Copy link to tab` has not the same format. It also seems strange to me that a link that works fine when opened via browser is not working with `executeDeepLink` (when talking about the link you can recover from Graph).

**Comment by Prasad-MSFT** [comment]
@Nootvar ,
Can you try as below:

```
var contentUrl = "xxxxxxxxxxxxxxxxxxxxxx";
var channelId = "19:7DSBv4H7fKAXrn5D72srpkMrvSwVqQb0qz7EKo6E9001@thread.tacv2";
microsoftTeams.executeDeepLink("https://9dc68128-88bf-4024-a275-7d049fbc7ac3/_djb2_msteams_prefix_96bbe562-5a47-4e15-a84f-0d2615883c7b?label=Informationdedossier&context={\"canvasUrl\":\"" + contentUrl + "\",\"channelId\":\"" + channelId + "\",\"subEntityId\":\"null\"}&groupId=7c541703-b4d3-4021-9281-51f1fe83713b&tenantId=b6c46f52-625c-414b-aca2-c4a4f0e7bd20");
```

**Comment by Nootvar** [comment]
Hello,

This open a new tab that tell me '9dc68128-88bf-4024-a275-7d049fbc7ac3’s server IP address could not be found'. I would have expected the link to contains `teams.microsoft.com/l/entity` so I tried to add it and got the same result as before (open the Teams Apps Catalog). Also, what is supposed to be the `contentUrl` value ? If there is an actual and important value for `contentUrl`, how can I obtain it programmatically ?

The tab I want to point at host an actual Team App that I can check out in the 'Manage apps' of my Team Administration (made trough 'Sync to Teams' in the App Catalog. That's why I don't really see what kind of value is expected by `contentUrl`.
